golang/pkg/dagent/utils/init_container.go

```diff
@@ -45,16 +51,14 @@ func spawnInitContainer(
 
 	resultCont, waitResult, err := builder.
 		WithClient(cli).
-		WithImage(cont.Image).
-		WithCmd(cont.Command).
+		WithImage(initCont.Config.Name).
+		WithCmd(initCont.Config.Command).
 		WithName(initContName).
-		WithEnv(EnvMapToSlice(cont.Envs)).
+		WithEnv(initCont.EnvList).
```

Comment:
We should use both InitContainer envs + parent config envs.

Reply:
Nice catch.